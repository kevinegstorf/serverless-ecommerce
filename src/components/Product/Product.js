import React, {Component} from "react";
import { css } from 'emotion';
import DefaultImage from '../content/DefaultImage/DefaultImage';
// import PayButton from "../PayButton";

const product = css`
  background-color: hotpink;
`

class Product extends Component {

  constructor(props) {
    super(props);

    this.state = {
      currentImage: 0,
    //   currentSku: props.skus[0]
    };
  }

  render() {
    const {id, name, caption, images, description} = this.props;
    const {currentImage} = this.state;

    const thumbnails = images.map((image, index) => {
      console.log(image);
      return image.length === 0 ? <DefaultImage/> : <img alt = 'product' key={index}
                  className={"product-thumbnail " + (index === currentImage
                      ? 'selected' : '')}
                  onClick={() => this.setState({currentImage: index})}
                  src={image} width="75"/>
    });

    const itemImage = (images) => {
      return images.length === 0 ? <DefaultImage/> : <img alt={name} className="product-image img-fluid"
      src={images[currentImage]}/>      
    }

    // const skuList = skus.map((sku, index) => {

    //   const {attributes} = sku;

    //   return (
    //       <div key={sku.id}
    //            className={"col-xs-2 product-sku " + (currentSku.id === sku.id
    //                ? 'selected' : '')}
    //            onClick={() => this.setState({currentSku: sku})}>
    //         {attributes.size}
    //       </div>
    //   );
    // });

    // const euros = price.substring(0, price.length - 2);
    // const cents = price.slice(-2);

    return (
        <div key={id} className={product}>
          <div className="product-images col-xs-12 col-sm-6 col-lg-8">
            <div className="product-image-wrapper row">
              <div className="col-xs-12 col-lg-2 d-none d-lg-block d-xl-block">
                <div className="product-thumbnails">{images.length === 0 ? <DefaultImage/> : thumbnails}</div>
              </div>
              <div className="col-xs-12 col-lg-10">
              {itemImage(images)}
              </div>
            </div>
          </div>
          <div className="product-details col-xs-12 col-sm-6 col-lg-4">
            <h2 className="product-name">{name}</h2>
            <h1 className="product-caption">{caption}</h1>
            <div
                className="product-price">
              <div className="product-taxes">Inclusief BTW</div>
            </div>
            <p className="product-description">{description}</p>
            <hr/>
            <hr/>
          </div>
        </div>
    );
  }
}
;

export default Product;