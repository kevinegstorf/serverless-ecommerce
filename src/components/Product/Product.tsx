import * as React from 'react';
import { css } from 'emotion';
import DefaultImage from '../DefaultImage/DefaultImage';
import {color} from '../../__styles__/index';
import placeholder from '../DefaultImage/images/default-image.svg';

console.log(color.pink)

const product = css`
  background-color: ${color.pink};
  font-family: 'Lato';
`

export type ProductProps = {
  id: string,
  name: string,
  caption: string,
  images: Array<string>,
  description: string,
  skus: string
}

class Product extends React.Component<ProductProps, any> {

  state = {
    currentImage: 0,
    defaultImage: placeholder
  };

  render() {
    const {id, name, caption, images, description} = this.props;
    const {currentImage} = this.state;

    const thumbnails = images.map((image:string, index: number) => {
      console.log(placeholder);
      return image.length === 0 ? <DefaultImage placeholder={this.state.defaultImage}/> : <img alt = 'product' key={index}
                  className={"product-thumbnail " + (index === currentImage
                      ? 'selected' : '')}
                  onClick={() => this.setState({currentImage: index})}
                  src={image} width="75"/>
    });

    const itemImage = (images: Array<string>) => {
      return images.length === 0 ? <DefaultImage placeholder={this.state.defaultImage}/> : <img alt={name} className="product-image img-fluid"
      src={images[currentImage]}/>      
    }
    return (
        <div key={id} className={product}>
          <div className="product-images col-xs-12 col-sm-6 col-lg-8">
            <div className="product-image-wrapper row">
              <div className="col-xs-12 col-lg-2 d-none d-lg-block d-xl-block">
                <div className="product-thumbnails">{images.length === 0 ? <DefaultImage placeholder={this.state.defaultImage}/> : thumbnails}</div>
              </div>
              <div className="col-xs-12 col-lg-10">
              {itemImage(images)}
              </div>
            </div>
          </div>
          <div className="product-details col-xs-12 col-sm-6 col-lg-4">
            <h2 className={product}>{name}</h2>
            <h1 className="product-caption">{caption}</h1>
            <div
                className="product-price">
              <div className="product-taxes">Inclusief BTW</div>
            </div>
            <p className="product-description">{description}</p>
            <hr/>
            <hr/>
          </div>
        </div>
    );
  }
}
;

export default Product;