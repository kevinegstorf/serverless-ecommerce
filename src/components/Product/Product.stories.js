import React from 'react';
import { storiesOf } from '@storybook/react';
import Product from './Product';

storiesOf('Product', module)
  .add('with text', () => <Product id={'test'} name={'name'} images={["https://s3-eu-west-1.amazonaws.com/aws-nodejs-typeâ€¦uck-94xqa2d8tkk1/images/white-t-shirt-500x500.jpg"]}/>);
