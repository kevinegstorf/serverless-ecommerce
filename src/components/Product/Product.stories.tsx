import React from 'react';
import { storiesOf } from '@storybook/react';
import { withState } from '@dump247/storybook-state';
import placeholder from '../DefaultImage/images/default-image.svg';


import Product from './Product';

(storiesOf('Product', module) as any)
  .addWithJSX('product', withState({ defaultImage: placeholder })(({ store }) => (
    <Product {...store.state} id='1' name='hello' caption='hello' description='test' images={[]} skus='test'/>
  )));
