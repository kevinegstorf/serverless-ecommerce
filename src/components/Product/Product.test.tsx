import React from 'react';
import { render } from 'react-testing-library';
import Product from './Product';

test('snapshopt test', () => {
    const container = render(<Product id='1' name='hello' caption='test' images={['hello']} description='test' skus='test'/>)
    expect(container).toMatchSnapshot();
});