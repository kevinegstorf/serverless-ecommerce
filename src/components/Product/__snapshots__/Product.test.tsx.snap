// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-uzmgpz"
      >
        <div
          class="product-images col-xs-12 col-sm-6 col-lg-8"
        >
          <div
            class="product-image-wrapper row"
          >
            <div
              class="col-xs-12 col-lg-2 d-none d-lg-block d-xl-block"
            >
              <div
                class="product-thumbnails"
              >
                <img
                  alt="product"
                  class="product-thumbnail selected"
                  src="hello"
                  width="75"
                />
              </div>
            </div>
            <div
              class="col-xs-12 col-lg-10"
            >
              <img
                alt="hello"
                class="product-image img-fluid"
                src="hello"
              />
            </div>
          </div>
        </div>
        <div
          class="product-details col-xs-12 col-sm-6 col-lg-4"
        >
          <h2
            class="product-name"
          >
            hello
          </h2>
          <h1
            class="product-caption"
          >
            test
          </h1>
          <div
            class="product-price"
          >
            <div
              class="product-taxes"
            >
              Inclusief BTW
            </div>
          </div>
          <p
            class="product-description"
          >
            test
          </p>
          <hr />
          <hr />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-uzmgpz"
    >
      <div
        class="product-images col-xs-12 col-sm-6 col-lg-8"
      >
        <div
          class="product-image-wrapper row"
        >
          <div
            class="col-xs-12 col-lg-2 d-none d-lg-block d-xl-block"
          >
            <div
              class="product-thumbnails"
            >
              <img
                alt="product"
                class="product-thumbnail selected"
                src="hello"
                width="75"
              />
            </div>
          </div>
          <div
            class="col-xs-12 col-lg-10"
          >
            <img
              alt="hello"
              class="product-image img-fluid"
              src="hello"
            />
          </div>
        </div>
      </div>
      <div
        class="product-details col-xs-12 col-sm-6 col-lg-4"
      >
        <h2
          class="product-name"
        >
          hello
        </h2>
        <h1
          class="product-caption"
        >
          test
        </h1>
        <div
          class="product-price"
        >
          <div
            class="product-taxes"
          >
            Inclusief BTW
          </div>
        </div>
        <p
          class="product-description"
        >
          test
        </p>
        <hr />
        <hr />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
